File: backend/Kinboard.Api/Models/JobAssignment.cs
namespace Kinboard.Api.Models;

public class JobAssignment
{
    public int Id { get; set; }
    
    // Foreign key to the parent Job
    public int JobId { get; set; }
    public Job Job { get; set; } = null!;
    
    // Foreign key to the assigned User
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    
    // Individual recurrence settings (used when Job.UseSharedRecurrence is false)
    public string? Recurrence { get; set; } // RRULE-like: FREQ=DAILY;INTERVAL=3 or FREQ=WEEKLY;INTERVAL=2;BYDAY=MO,TH,SU
    public DateTime? RecurrenceStartDate { get; set; }
    public DateTime? RecurrenceEndDate { get; set; }
    public bool RecurrenceIndefinite { get; set; }
    
    // Display order for this assignment within the job
    public int DisplayOrder { get; set; } = 0;
    
    // Navigation property for completions specific to this assignment
    public ICollection<JobCompletion> Completions { get; set; } = new List<JobCompletion>();
}

File: kinboard-tv/app/src/main/java/com/kinboard/tv/data/model/JobAssignment.kt
package com.kinboard.tv.data.model

import com.google.gson.annotations.SerializedName

data class JobAssignment(
    @SerializedName("id")
    val id: Int,
    
    @SerializedName("jobId")
    val jobId: Int,
    
    @SerializedName("userId")
    val userId: Int,
    
    @SerializedName("user")
    val user: User? = null,
    
    @SerializedName("recurrence")
    val recurrence: String? = null,
    
    @SerializedName("recurrenceStartDate")
    val recurrenceStartDate: String? = null,
    
    @SerializedName("recurrenceEndDate")
    val recurrenceEndDate: String? = null,
    
    @SerializedName("recurrenceIndefinite")
    val recurrenceIndefinite: Boolean? = null,
    
    @SerializedName("displayOrder")
    val displayOrder: Int,
    
    @SerializedName("isCompleted")
    val isCompleted: Boolean? = null,
    
    @SerializedName("completedAt")
    val completedAt: String? = null  // ISO datetime
)
